controllers/limitador_controller.go

```diff
@@ -111,22 +110,108 @@ func (r *LimitadorReconciler) SetupWithManager(mgr ctrl.Manager) error {
 
 func (r *LimitadorReconciler) reconcileStatus(ctx context.Context, limitadorObj *limitadorv1alpha1.Limitador) error {
 	logger := logr.FromContext(ctx)
-	// Simple enough now, we could implement a thorough check like with RateLimit with ObservedGeneration in the future
-	builtServiceUrl := buildServiceUrl(limitadorObj)
-	if builtServiceUrl != limitadorObj.Status.ServiceURL {
-		logger.V(1).Info("Updating the Status", "Name", limitadorObj.Name)
-		limitadorObj.Status.ServiceURL = builtServiceUrl
-		return r.Client().Status().Update(ctx, limitadorObj)
 
+	isLimitadorRunning := r.checkLimitadorInstanceIsRunning(ctx, limitadorObj)
+	changed := updateStatusReady(limitadorObj, isLimitadorRunning)
+
+	changed = updateStatusService(limitadorObj) || changed
+
+	if !limitadorObj.Status.Ready() {
+		return fmt.Errorf("resource not ready")
```

Comment:
Now the reconciliation logic will re-queue until the Deployment is ready. If the Deployment for some reason has failed, it will be reflected the error in the logs. If needed, I could refactor de logic in the reconciliation when there's a fail in the deployment and stop the re-queuing ?